pandas/core/reshape/merge.py

```diff
@@ -109,6 +109,7 @@
     np.int64: libhashtable.Int64Factorizer,
     np.longlong: libhashtable.Int64Factorizer,
     np.int32: libhashtable.Int32Factorizer,
+    np.intc: libhashtable.Int32Factorizer,
```

Comment:
is this going to be platform-dependent?

Reply:
As far as I can see, an actual `np.intc` is only available on Windows, and for Linux (and probably also Mac OS) it is `np.int32`. 

TBH, I did not know about `np.intc` before some tests started failing.

Reply:
Do we have a generic IntFactorizer? It is a bit pedantic since I think most distributions will have an int be 32 bits, but that definitely is not a guarantee

Reply:
iirc @seberg mentioned hoping to get rid of np.intc in numpy 2.0

Reply:
I am not worried about `intc`, bu its system dependend, it doesn't have to be 32bit, though.  `cnp.int_t` is what worries me a bit, because its `long`, but if we want 64bit on windows it won't match up with `np.array([1])` anymore.

Reply:
I assume this regression came as part of https://github.com/pandas-dev/pandas/pull/49876
To avoid going down a rabbit hole I would be OK with just setting this specific case back to the Int64Factorizer. @phofl